tests/petsc/assemble_matrix_parallel_01.cc

```diff
@@ -0,0 +1,493 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2009 - 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+
+// same as deal.II/assemble_matrix_parallel_01, but for PETSc matrices
+// and vectors
+//
+// This test requires PETSc to be configured with the option
+// "--with-threadsafety" in case of debug builds of PETSc
+// For optimized builds, the above option is needed only in case
+// users want PETSc to produce useful logs with "-log_view" runtime
+// option
+#include <deal.II/base/function.h>
+#include <deal.II/base/graph_coloring.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/utilities.h>
+#include <deal.II/base/work_stream.h>
+
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+
+#include <deal.II/hp/fe_values.h>
+
+#include <deal.II/lac/affine_constraints.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/petsc_sparse_matrix.h>
+#include <deal.II/lac/vector.h>
+
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/vector_tools.h>
+
+#include <complex>
+#include <iostream>
+
+#include "../tests.h"
+
+std::ofstream logfile("output");
```

Comment:
We have a lot of tests that do this but we are (very slowly) trying to replace them with `initlog()` or `mpi_initlog()` - could you switch to using that for this new test?

Reply:
I just copy-pasted the equivalent test for Trilinos and changed the classes. Why don't apply these changes in a single large commit to all tests at once?

Reply:
@drwells did that in #14719 now. Would you mind following the style used there?

Reply:
done